### PR TITLE
DEVPROD-2643: Fix useLogDownloader bug

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-2643

### Description
- First commit reverts #434 i. e. reintroduces #431 
- Second commit fixes bug
  - Remove useLogDownloader dependency on isEvergreenLoading. This dependency caused the hook to run multiple times; when the bug was deployed to production you could observe multiple requests to Logkeeper. I think the subsequent request to Logkeeper also resulted in the first one being terminated, hence the "Incomplete log" warnings we saw earlier today.
  - In the case of an invalid task URL (i.e. `/evergreen/invalid/0/task`) where the GraphQL query doesn't return a download link, use the previous approach to construct a link given the task ID, execution, etc.

### Testing
- Verified on beta that removing this dependency results in a single call to fetch the log. It's deployed to beta right now if you care to look!

### CODE REVIEW

File: src/constants/logURLTemplates.ts [hunk 1]
```diff
@@ -101,40 +103,26 @@ const getResmokeLogURL = (
...
- * @returns an Evergreen URL of the format `/task/${taskID}/${execution}?type=${OriginToType[origin]}&text=true`
- */
 const getEvergreenTaskLogURL = (
-  taskID: string,
-  execution: string | number,
-  origin: keyof typeof OriginToType,
-  options: { priority?: boolean; text?: boolean }
+  logLinks: TaskType["logs"],
+  origin: string,
+  params: { [key: string]: any } = {}
```
Comment (minnakt) [inline src/constants/logURLTemplates.ts:116]: does it make sense to type the params a bit more strongly?

File: src/hooks/useLogDownloader/index.ts [hunk 2]
```diff
@@ -17,28 +17,35 @@ import {
 import { fetchLogFile } from "utils/fetchLogFile";
```
Comment (minnakt) [inline src/hooks/useLogDownloader/index.ts:17]: it's not a big deal, but do you know where this flash of 0 comes from? (only happens on bigger logs)

https://github.com/evergreen-ci/parsley/assets/47064971/5a98864c-3cdb-4eb2-babd-c492171cbdb3

File: src/hooks/useLogDownloader/index.ts [hunk 3]
```diff
@@ -142,6 +150,8 @@ const useLogDownloader = (
           });
           setIsLoading(false);
         });
+    } else if (url === null) {
+      setError("Log URL not specified, unable to download.");
```
Comment (minnakt) [inline src/hooks/useLogDownloader/index.ts:154]: I think i might tweak the wording here to something along the lines of this:
```suggestion
      setError("Could not retrieve log URL from task, unable to download.");
```

**Review by minnakt — COMMENTED** [review]
sorry for the late review 😭 pls ping me directly whenever you want a re-review!

File: src/constants/logURLTemplates.ts (hunk 1 above)
Comment (sophstad) [reply src/constants/logURLTemplates.ts:116]: This seemed to be the general typing we use for query params, à la `useQueryParams` https://github.com/evergreen-ci/parsley/blob/4126a3928d37da614d211fb0ecb3ef2735ac0ab6/src/hooks/useQueryParam/index.ts#L17

File: src/hooks/useLogDownloader/index.ts (hunk 2 above)
Comment (sophstad) [reply src/hooks/useLogDownloader/index.ts:17]: I actually think this is a React strict-mode quirk that occurs locally and results in certain things firing twice! We've been running into more of those lately. I can reproduce this on `main` when running locally, and don't see the behavior when my branch is deployed to beta.

File: src/pages/LogView/LoadingPage/useResolveLogURL.test.tsx [hunk 4]
```diff
@@ -129,6 +175,7 @@ describe("useResolveLogURL", () => {
       }
     );
     await waitForNextUpdate();
+    await waitForNextUpdate();
```
[the remark this answers is not in the document]
Comment (sophstad) [reply src/pages/LogView/LoadingPage/useResolveLogURL.test.tsx:178]: My mistake thank you!

File: src/pages/LogView/LoadingPage/useResolveLogURL.ts [hunk 5]
```diff
@@ -142,19 +151,32 @@ export const useResolveLogURL = ({
...
+            text: true,
+          })
+        : constructEvergreenTaskLogURL(taskID, execution, origin as any, {
+            text: true,
+          });
+      htmlLogURL = task?.logs
+        ? getEvergreenTaskLogURL(task.logs, origin, {
+            text: false,
+          })
+        : constructEvergreenTaskLogURL(taskID, execution, origin as any, {
```
Comment (minnakt) [inline src/pages/LogView/LoadingPage/useResolveLogURL.ts:177]: wondering if it makes sense to copy the types for `getEvergreenTaskLogURL` to remove the `origin as any` stuff!

File: src/pages/LogView/LoadingPage/useResolveLogURL.test.tsx [hunk 6]
```diff
@@ -184,7 +254,7 @@ const getEmptyTestLogURLMock: ApolloMock<
     variables: {
       execution: 0,
       taskID: "a-task-id",
-      testName: "^a-test-name$",
+      testName: "^a-test-name-that-doesnt-exist$",
```
Comment (minnakt) [inline src/pages/LogView/LoadingPage/useResolveLogURL.test.tsx:257]: ty for fixing 😁

**Review by minnakt — APPROVED** [review]
lgtm!

**Pull request merged**